app/javascript/components/explore/ExploreDisplayPanelManager.jsx

```diff
@@ -0,0 +1,473 @@
+      cluster_name: annot.cluster_name,
+      name: annot.annotation_name,
+      scope: annot.annotation_scope,
+      type: 'group'
+    }
+  })
+
+  const clustersWithDe = Array.from(new Set(annotsWithDe.map(a => a.cluster_name)))
+
+  return {
+    clusters: clustersWithDe,
+    annotations: annotsWithDe,
+    subsample_thresholds: exploreInfo.annotationList.subsample_thresholds
+  }
+}
+
+
+
+/** Determine if current annotation has differential expression results available */
+function getAnnotHasDe(exploreInfo, exploreParams) {
+  const flags = getFeatureFlagsWithDefaults()
+  if (!flags?.differential_expression_frontend || !exploreInfo) {
+    // set isDifferentialExpressionEnabled to false as user cannot see DE results, even if present for annotation
+    if (window.SCP) {
+      window.SCP.isDifferentialExpressionEnabled = false
+    }
+    return false
+  }
+
+  let annotHasDe = false
+  const [selectedCluster, selectedAnnot] = getSelectedClusterAndAnnot(exploreInfo, exploreParams)
+
+  annotHasDe = exploreInfo.differentialExpression.some(deItem => {
+    return (
+      deItem.cluster_name === selectedCluster &&
+      deItem.annotation_name === selectedAnnot.name &&
+      deItem.annotation_scope === selectedAnnot.scope
+    )
+  })
+
+  return annotHasDe
+}
+
+/**
+ * Determine if current annotation has differential expression results that are user-generated.
+ *
+ * DE results have two dimensions:
+ * - Comparison type: either "one-vs-rest" or "pairwise"
+ * - Source: "author-computed" "or SCP-computed"
+ *
+ * Author-computed DE is also often called "precomputed" or "user-uploaded" or "study-owner-generated"
+ * or "custom".  Whereas SCP-generated DE is computed only for cell-type-like annotations and only as
+ * one-vs-rest comparisons, user-generated DE can be more comprehensive -- it can be available for
+ * any annotation, and as one-vs-rest and/or pairwise comparisons.
+ */
+function getIsAuthorDe(exploreInfo, exploreParams) {
+  const flags = getFeatureFlagsWithDefaults()
+  if (!flags?.differential_expression_frontend || !exploreInfo) {
+    return false
+  }
+
+  const [selectedCluster, selectedAnnot] = getSelectedClusterAndAnnot(exploreInfo, exploreParams)
+
+  const deItem = exploreInfo.differentialExpression.find(deItem => {
+    return (
+      deItem.cluster_name === selectedCluster &&
+      deItem.annotation_name === selectedAnnot.name &&
+      deItem.annotation_scope === selectedAnnot.scope
+    )
+  })
+
+  const isAuthorDe = deItem?.select_options.is_author_de
+
+  return isAuthorDe
+}
+
+/**
+ * Manages the right panel section of the explore view of a study. We have three options for the right side
+ * panel with different controls to adjust the plots: Options (or default), Differential Expression, and
+ * Facet Filtering (currently in progress)
+ *
+ * This manager will return the correct content and header for the appropriate panel to display.
+ *
+ *  */
+export default function ExploreDisplayPanelManager({
```

Comment:
I totally get why we have to do this, but I get slightly worried at the amount of arguments for this function.  30 seems unwieldy, but having said that I don't see how we get around that without more refactoring.

Reply:
Yeah 100% agree. I hope to reduce this in my next PR as I keep working out the stuff actually necessary to be shared between the components.